## 1. Data Ingestion

In [ ]:
# pip install pandas

In [1]:
import pandas as pd

# Load Ghana data
ghana = pd.read_csv('C:\\Users\\Surface\\OneDrive\\Documentos\\GitHub\\sage-hypertension-predictor-ssa\\data\\31381-0004-Data.tsv', sep='\t', low_memory=False)

# Load South Africa data
south_africa = pd.read_csv('C:\\Users\\Surface\\OneDrive\\Documentos\\GitHub\\sage-hypertension-predictor-ssa\\data\\31381-0012-Data.tsv', sep='\t', low_memory=False)

# Check shapes
print("Ghana shape:", ghana.shape)
print("South Africa shape:", south_africa.shape)

# Preview
ghana.head()

Ghana shape: (5573, 1677)
South Africa shape: (4227, 1677)


,id,id2ndwave,q0001,q0002,q0002a,q0003,q0008,q0101b,q0104,q0105b,...,q9009,q9010,q9011,q9012,q9013,q9014,strata,pweight,income,quintile_c
0,2410000101,24100001101,241,1,2,1,A,515018462,2,GHA004,...,2,2,1,2,3,2,52,1480.63830566406,.705987691879272,5
1,2410000201,24100002101,241,2,2,2,D,602006412,2,GHA001,...,2,2,2,2,1,1,62,1071.89599609375,.592056274414062,4
2,2410000504,24100005104,241,5,2,2,B,103021453,2,GHA010,...,2,2,2,2,1,1,12,320.957672119141,.845640897750854,5
3,2410000801,24100008101,241,8,1,2,B,103021453,2,GHA010,...,2,2,2,2,1,3,12,6868.7783203125,.707030296325684,5
4,2410001201,24100012101,241,12,2,1,C,103021453,2,GHA010,...,2,2,2,2,1,1,12,407.469635009766,.602124810218811,4


In [2]:
# Check column names
print(ghana.columns.tolist())

['id', 'id2ndwave', 'q0001', 'q0002', 'q0002a', 'q0003', 'q0008', 'q0101b', 'q0104', 'q0105b', 'q0404a', 'q0404b', 'q0405', 'q0406', 'q0407', 'q0408', 'q0409', 'q0411', 'q0412', 'q0413', 'q0414', 'q0415', 'q0416', 'q1000a', 'q1000b', 'q1000c', 'q1000cactual', 'q1001', 'q1002', 'q1003', 'q1004', 'q1005', 's1000_hh_b', 's1000_mm_b', 's1000_hh_e', 's1000_mm_e', 'q1006', 'q1007', 'q1008', 'q1009', 'q1010_mm', 'q1010_yyyy', 'q1011', 'q1012', 'q1013', 'q1014', 'q1015', 'q1016', 'q1017', 'q1018', 'q1019', 'q1020', 'q1021', 'q1022', 'q1023', 'q1024', 'q1025', 'q1026', 'q1027', 'q1028', 'q1029', 'q1030', 'q1031', 'q1032', 's1500_hh_b', 's1500_mm_b', 's1500_hh_e', 's1500_mm_e', 'q1501', 'q1501a', 'q1502', 'q1502a', 'q1503', 'q1504', 'q1505', 'q1505a', 'q1506', 'q1507', 'q1508', 'q1509', 'q1510', 'q1511', 'q1512', 'q1513', 'q1514a', 'q1514b', 'q1514c', 'q1514d', 'q1514e', 'q1514f', 'q1515', 's2000_hh_b', 's2000_mm_b', 's2000_hh_e', 's2000_mm_e', 'q2000', 'q2001', 'q2002', 'q2003', 'q2004', 'q2005

## 2. Data Familiarization

In [3]:
## Selecting Relevant Columns Using the codebook
# Select relevant variables
cols = [
    'q0001', 'q1009', 'q1010_mm', 'q1010_yyyy', 'q0104',
    'q2501_s', 'q2501_d', 'q2502_s', 'q2502_d', 'q2503_s', 'q2503_d',
    'q2506', 'q2507', 'q2508', 'q2509',
    'q3001', 'q3002', 'q3007', 'q3008', 'q3011',
    'q3016', 'q3017', 'q3019', 'q3022', 'q3025', 'q3028',
    'q3012', 'q3013', 'pweight'
]

# South Africa cols - birth year removed (privacy restriction)
cols_sa = [c for c in cols if c != 'q1010_yyyy']

ghana_clean = ghana[cols].copy()
south_africa_clean = south_africa[cols_sa].copy()

print(ghana_clean.shape)
print(south_africa_clean.shape)

(5573, 29)
(4227, 28)


In [4]:
# Check missing values - Ghana
print(ghana_clean.isnull().sum())
# Check data types - Ghana
print(ghana_clean.dtypes)
# Check missing values - south Africa 
print(south_africa_clean.isnull().sum())
# check data types - South Africa
print(south_africa_clean.dtypes)

q0001         0
q1009         0
q1010_mm      0
q1010_yyyy    0
q0104         0
q2501_s       0
q2501_d       0
q2502_s       0
q2502_d       0
q2503_s       0
q2503_d       0
q2506         0
q2507         0
q2508         0
q2509         0
q3001         0
q3002         0
q3007         0
q3008         0
q3011         0
q3016         0
q3017         0
q3019         0
q3022         0
q3025         0
q3028         0
q3012         0
q3013         0
pweight       0
dtype: int64
q0001         int64
q1009           str
q1010_mm        str
q1010_yyyy      str
q0104           str
q2501_s         str
q2501_d         str
q2502_s         str
q2502_d         str
q2503_s         str
q2503_d         str
q2506           str
q2507           str
q2508           str
q2509           str
q3001           str
q3002           str
q3007           str
q3008           str
q3011           str
q3016           str
q3017           str
q3019           str
q3022           str
q3025           str
q3028           str
q30

In [5]:
# Check unique values in systolic BP column
print(ghana_clean['q2501_s'].unique()[:30])

<StringArray>
['189', '187', '140', '122', '152', '105',  '89', '142', '204', '128', '215',
 '136', '123', '143', '160', '108', '162', '120',   ' ', '106', '101', '114',
 '117', '132', '119', '202', '133', '161', '116', '129']
Length: 30, dtype: str


In [6]:
# Count blank spaces in BP columns
bp_cols = ['q2501_s', 'q2501_d', 'q2502_s', 'q2502_d', 'q2503_s', 'q2503_d']

for col in bp_cols:
    blanks = (ghana_clean[col].str.strip() == '').sum()
    print(f"Ghana {col}: {blanks} blanks")

print()

for col in bp_cols:
    blanks = (south_africa_clean[col].str.strip() == '').sum()
    print(f"South Africa {col}: {blanks} blanks")

Ghana q2501_s: 475 blanks
Ghana q2501_d: 475 blanks
Ghana q2502_s: 475 blanks
Ghana q2502_d: 475 blanks
Ghana q2503_s: 475 blanks
Ghana q2503_d: 475 blanks

South Africa q2501_s: 92 blanks
South Africa q2501_d: 92 blanks
South Africa q2502_s: 93 blanks
South Africa q2502_d: 93 blanks
South Africa q2503_s: 93 blanks
South Africa q2503_d: 93 blanks


In [7]:
# Check blanks in all other columns
other_cols = ['q2506', 'q2507', 'q2508', 'q2509', 
              'q1009', 'q0104', 'q3001', 'q3002',
              'q3007', 'q3008', 'q3011', 'q3016',
              'q3019', 'q3022', 'q3012', 'q3013']

print("GHANA")
for col in other_cols:
    blanks = (ghana_clean[col].str.strip() == '').sum()
    print(f"{col}: {blanks} blanks")

print("\nSOUTH AFRICA")
for col in other_cols:
    blanks = (south_africa_clean[col].str.strip() == '').sum()
    print(f"{col}: {blanks} blanks")

GHANA
q2506: 572 blanks
q2507: 572 blanks
q2508: 572 blanks
q2509: 572 blanks
q1009: 8 blanks
q0104: 2 blanks
q3001: 481 blanks
q3002: 4330 blanks
q3007: 481 blanks
q3008: 2584 blanks
q3011: 3224 blanks
q3016: 481 blanks
q3019: 481 blanks
q3022: 481 blanks
q3012: 481 blanks
q3013: 481 blanks

SOUTH AFRICA
q2506: 165 blanks
q2507: 166 blanks
q2508: 166 blanks
q2509: 167 blanks
q1009: 2 blanks
q0104: 6 blanks
q3001: 189 blanks
q3002: 2786 blanks
q3007: 190 blanks
q3008: 3112 blanks
q3011: 3178 blanks
q3016: 190 blanks
q3019: 190 blanks
q3022: 190 blanks
q3012: 190 blanks
q3013: 190 blanks


In [8]:
# Convert and check BP ranges
ghana_clean['q2501_s_num'] = pd.to_numeric(ghana_clean['q2501_s'], errors='coerce')
print(ghana_clean['q2501_s_num'].describe())

count    5098.000000
mean      136.810514
std        31.112077
min        -8.000000
25%       120.000000
50%       135.000000
75%       153.000000
max       240.000000
Name: q2501_s_num, dtype: float64


In [9]:
print(ghana_clean['q1009'].unique())

<StringArray>
['2', '1', ' ']
Length: 3, dtype: str


In [10]:
print(ghana_clean['q1010_yyyy'].unique()[:20])

<StringArray>
['1947',   '-8', '1964', '1966', '1988', '1967', '1973',    ' ', '1957',
 '1986', '1965', '1975', '1916', '1972', '1940', '1936', '1951', '1959',
 '1953', '1984']
Length: 20, dtype: str


In [11]:
print(ghana_clean['q1009'].value_counts())

q1009
1    2816
2    2749
        8
Name: count, dtype: int64


In [12]:
# Check for -8 codes across all numeric-ish columns
cols_to_check = ['q2501_s', 'q2501_d', 'q2502_s', 'q2502_d', 
                 'q2503_s', 'q2503_d', 'q2506', 'q2507']

for col in cols_to_check:
    count = (ghana_clean[col].str.strip() == '-8').sum()
    print(f"Ghana {col}: {count} values of -8")

Ghana q2501_s: 68 values of -8
Ghana q2501_d: 79 values of -8
Ghana q2502_s: 73 values of -8
Ghana q2502_d: 87 values of -8
Ghana q2503_s: 86 values of -8
Ghana q2503_d: 96 values of -8
Ghana q2506: 0 values of -8
Ghana q2507: 0 values of -8


In [13]:
# Check a few rows of sex vs other variables to infer coding
ghana_clean[['q1009']].value_counts()

q1009
1        2816
2        2749
            8
Name: count, dtype: int64

## Intepretation

##### Dataset Shape
- Ghana: 5,573 respondents, 29 variables selected
- South Africa: 4,227 respondents, 28 variables selected

##### Variable Coding
- Sex: 1 = Male, 2 = Female
- Missing values encoded as blank spaces ' '
- Invalid/unknown values encoded as -8

##### Missing Data
- BP readings: 475 Ghana (8.5%), 92-93 South Africa (2.2%) — will drop
- Height/weight: 572 Ghana, ~166 South Africa — will impute or drop
- Lifestyle variables: ~481 Ghana, ~190 South Africa missing entire section
- q3002, q3008, q3011 high blanks are expected skip logic, not missing data

##### Data Quality
- BP range: min -8 (invalid), max 240 mmHg — clinically plausible after cleaning
- Birth years: 1916-1988 — reasonable for 18+ adults
- -8 codes in BP columns: 68-96 per column in Ghana — treat as missing

##### Preprocessing Plan
1. Replace blank spaces and -8 with NaN
2. Drop rows missing BP readings
3. Convert string columns to numeric
4. Calculate age from birth year
5. Combine Ghana and South Africa into one dataset

## Preprocessing 

In [14]:
print("Ghana:", ghana.shape)
print("South Africa:", south_africa.shape)

Ghana: (5573, 1677)
South Africa: (4227, 1677)


In [15]:
cols = [
    'id', 'q0001', 'q1009', 'q1010_mm', 'q1010_yyyy', 'q0104',
    'q1011',
    'q2501_s', 'q2501_d', 'q2502_s', 'q2502_d', 'q2503_s', 'q2503_d',
    'q2506', 'q2507', 'q2508', 'q2509',
    'q3001', 'q3002', 'q3007', 'q3008', 'q3011',
    'q3016', 'q3017', 'q3019', 'q3022', 'q3025', 'q3028',
    'q3012', 'q3013', 'pweight'
]

cols_sa = [c for c in cols if c != 'q1010_yyyy']

ghana_clean = ghana[cols].copy()
south_africa_clean = south_africa[cols_sa].copy()

print("Ghana:", ghana_clean.shape)
print("South Africa:", south_africa_clean.shape)

Ghana: (5573, 31)
South Africa: (4227, 30)


In [16]:
import numpy as np

ghana_clean = ghana_clean.replace(' ', np.nan)
ghana_clean = ghana_clean.replace('-8', np.nan)

south_africa_clean = south_africa_clean.replace(' ', np.nan)
south_africa_clean = south_africa_clean.replace('-8', np.nan)

print("Done")

Done


In [17]:
bp_cols = ['q2501_s', 'q2502_s', 'q2503_s', 'q2501_d', 'q2502_d', 'q2503_d']

ghana_clean = ghana_clean.dropna(subset=bp_cols)
south_africa_clean = south_africa_clean.dropna(subset=bp_cols)

print("Ghana:", ghana_clean.shape)
print("South Africa:", south_africa_clean.shape)

Ghana: (4966, 31)
South Africa: (4020, 30)


In [18]:
numeric_cols = [
    'q2501_s', 'q2501_d', 'q2502_s', 'q2502_d', 'q2503_s', 'q2503_d',
    'q2506', 'q2507', 'q2508', 'q2509',
    'q1010_mm', 'q1010_yyyy', 'q1011',
    'q3001', 'q3002', 'q3007', 'q3008', 'q3011',
    'q3016', 'q3017', 'q3019', 'q3022', 'q3025', 'q3028',
    'q3012', 'q3013', 'pweight'
]

for col in numeric_cols:
    if col in ghana_clean.columns:
        ghana_clean[col] = pd.to_numeric(ghana_clean[col], errors='coerce')
    if col in south_africa_clean.columns:
        south_africa_clean[col] = pd.to_numeric(south_africa_clean[col], errors='coerce')

print(ghana_clean.dtypes)

id              int64
q0001           int64
q1009             str
q1010_mm      float64
q1010_yyyy    float64
q0104             str
q1011         float64
q2501_s         int64
q2501_d         int64
q2502_s         int64
q2502_d         int64
q2503_s         int64
q2503_d         int64
q2506         float64
q2507         float64
q2508         float64
q2509         float64
q3001         float64
q3002         float64
q3007         float64
q3008         float64
q3011         float64
q3016         float64
q3017         float64
q3019         float64
q3022         float64
q3025         float64
q3028         float64
q3012         float64
q3013         float64
pweight       float64
dtype: object


In [19]:
skip_cols = ['q3002', 'q3008', 'q3011', 'q3017']

for col in skip_cols:
    ghana_clean[col] = ghana_clean[col].fillna(0)
    south_africa_clean[col] = south_africa_clean[col].fillna(0)

print("Done")

Done


In [20]:
small_null_cols = ['q3001', 'q3007', 'q3016', 'q3019', 
                   'q3022', 'q3025', 'q3028']

ghana_clean = ghana_clean.dropna(subset=small_null_cols)
south_africa_clean = south_africa_clean.dropna(subset=small_null_cols)

print("Ghana:", ghana_clean.shape)
print("South Africa:", south_africa_clean.shape)

Ghana: (4961, 31)
South Africa: (3931, 30)


In [21]:
body_cols = ['q2506', 'q2507', 'q2508', 'q2509']

for col in body_cols:
    ghana_clean[col] = ghana_clean[col].fillna(ghana_clean[col].median())
    south_africa_clean[col] = south_africa_clean[col].fillna(south_africa_clean[col].median())

print("Done")

Done


In [22]:
# Diet columns - median
for col in ['q3012', 'q3013']:
    ghana_clean[col] = ghana_clean[col].fillna(ghana_clean[col].median())
    south_africa_clean[col] = south_africa_clean[col].fillna(south_africa_clean[col].median())

# Sex and residence - mode
south_africa_clean['q1009'] = south_africa_clean['q1009'].fillna(
    south_africa_clean['q1009'].mode()[0])
south_africa_clean['q0104'] = south_africa_clean['q0104'].fillna(
    south_africa_clean['q0104'].mode()[0])

# Survey weight - drop
ghana_clean = ghana_clean.dropna(subset=['pweight'])
south_africa_clean = south_africa_clean.dropna(subset=['pweight'])

print("Ghana:", ghana_clean.shape)
print("South Africa:", south_africa_clean.shape)

Ghana: (4960, 31)
South Africa: (3927, 30)


In [23]:
rename_cols = {
    'id': 'id',
    'q0001': 'country',
    'q1009': 'sex',
    'q1010_mm': 'birth_month',
    'q1010_yyyy': 'birth_year',
    'q0104': 'residence',
    'q1011': 'age',
    'q2501_s': 'sbp1', 'q2501_d': 'dbp1',
    'q2502_s': 'sbp2', 'q2502_d': 'dbp2',
    'q2503_s': 'sbp3', 'q2503_d': 'dbp3',
    'q2506': 'height_cm', 'q2507': 'weight_kg',
    'q2508': 'waist_cm', 'q2509': 'hip_cm',
    'q3001': 'ever_tobacco', 'q3002': 'current_tobacco',
    'q3007': 'ever_alcohol', 'q3008': 'recent_alcohol',
    'q3011': 'avg_drinks_daily', 'q3016': 'vigorous_work',
    'q3017': 'days_vigorous_work', 'q3019': 'moderate_work',
    'q3022': 'walks_or_cycles', 'q3025': 'vigorous_leisure',
    'q3028': 'moderate_leisure', 'q3012': 'fruit_servings',
    'q3013': 'veg_servings', 'pweight': 'survey_weight'
}

ghana_clean = ghana_clean.rename(columns=rename_cols)
south_africa_clean = south_africa_clean.rename(columns=rename_cols)

print(ghana_clean.columns.tolist())

['id', 'country', 'sex', 'birth_month', 'birth_year', 'residence', 'age', 'sbp1', 'dbp1', 'sbp2', 'dbp2', 'sbp3', 'dbp3', 'height_cm', 'weight_kg', 'waist_cm', 'hip_cm', 'ever_tobacco', 'current_tobacco', 'ever_alcohol', 'recent_alcohol', 'avg_drinks_daily', 'vigorous_work', 'days_vigorous_work', 'moderate_work', 'walks_or_cycles', 'vigorous_leisure', 'moderate_leisure', 'fruit_servings', 'veg_servings', 'survey_weight']


In [24]:
south_africa_clean['birth_year'] = np.nan

df = pd.concat([ghana_clean, south_africa_clean], ignore_index=True)

print("Combined dataset:", df.shape)
print(df['country'].value_counts())

Combined dataset: (8887, 31)
country
241    4960
155    3927
Name: count, dtype: int64


In [25]:
print(df.isnull().sum())

id                       0
country                  0
sex                      0
birth_month           4015
birth_year            7056
residence                0
age                      0
sbp1                     0
dbp1                     0
sbp2                     0
dbp2                     0
sbp3                     0
dbp3                     0
height_cm                0
weight_kg                0
waist_cm                 0
hip_cm                   0
ever_tobacco             0
current_tobacco          0
ever_alcohol             0
recent_alcohol           0
avg_drinks_daily         0
vigorous_work            0
days_vigorous_work       0
moderate_work            0
walks_or_cycles          0
vigorous_leisure         0
moderate_leisure         0
fruit_servings           0
veg_servings             0
survey_weight            0
dtype: int64


In [26]:
df.to_csv('../docs/df_preprocessed.csv', index=False)
print("Saved successfully")

Saved successfully


## Preprocessing Summary

**Input:** Ghana (5,573) + South Africa (4,227) = 9,800 respondents
**Output:** Combined clean dataset of 8,777 respondents, 31 variables

### Steps Performed

1. Replaced blank spaces and -8 codes with NaN
2. Dropped rows missing blood pressure readings (6.2% of data)
3. Converted all string columns to numeric types
4. Filled skip logic nulls with 0 (current_tobacco, recent_alcohol,
   avg_drinks_daily, days_vigorous_work)
5. Imputed body measurements and diet variables with column median
6. Imputed sex and residence with column mode
7. Dropped rows with missing survey_weight
8. Added id and q1011 (direct age in years) to variable selection
9. Added birth_year as NaN to South Africa to maintain structure
10. Combined Ghana and South Africa into single dataframe

### Remaining Nulls
- birth_month and birth_year remain but are no longer needed
- Direct age variable (q1011) covers all 8,777 respondents

### Country Codes
- 241 = Ghana (4,960 respondents)
- 155 = South Africa (3,927 respondents)

### Key Decisions
- BP missing rows dropped — target variable cannot be imputed
- Skip logic nulls filled with 0 — absence of behaviour is a real value
- Median used over mean for body measurements — resistant to outliers
- Row count dropped from 9,800 to 8,777 — 10.4% total data loss,
  within acceptable limits for health research

## Day 6. Feature Engineering

In [27]:
df = pd.read_csv('../docs/df_preprocessed.csv')
print(df.shape)
print(df.columns.tolist())

(8887, 31)
['id', 'country', 'sex', 'birth_month', 'birth_year', 'residence', 'age', 'sbp1', 'dbp1', 'sbp2', 'dbp2', 'sbp3', 'dbp3', 'height_cm', 'weight_kg', 'waist_cm', 'hip_cm', 'ever_tobacco', 'current_tobacco', 'ever_alcohol', 'recent_alcohol', 'avg_drinks_daily', 'vigorous_work', 'days_vigorous_work', 'moderate_work', 'walks_or_cycles', 'vigorous_leisure', 'moderate_leisure', 'fruit_servings', 'veg_servings', 'survey_weight']


In [28]:
df['sbp_avg'] = df[['sbp1', 'sbp2', 'sbp3']].mean(axis=1)
df['dbp_avg'] = df[['dbp1', 'dbp2', 'dbp3']].mean(axis=1)

# Fix -9 codes that slipped through
df['sbp_avg'] = df['sbp_avg'].replace(-9, np.nan)
df['dbp_avg'] = df['dbp_avg'].replace(-9, np.nan)
df = df.dropna(subset=['sbp_avg', 'dbp_avg'])

print(df['sbp_avg'].describe())

count    8886.000000
mean      140.118651
std        25.239203
min        68.666667
25%       121.666667
50%       136.666667
75%       155.333333
max       233.000000
Name: sbp_avg, dtype: float64


In [29]:
def classify_bp(row):
    sbp = row['sbp_avg']
    dbp = row['dbp_avg']
    if sbp >= 180 or dbp >= 110:
        return 3  # Grade 3
    elif sbp >= 160 or dbp >= 100:
        return 2  # Grade 2
    elif sbp >= 140 or dbp >= 90:
        return 1  # Grade 1
    else:
        return 0  # Normal

df['hypertension_grade'] = df.apply(classify_bp, axis=1)
print(df['hypertension_grade'].value_counts())

hypertension_grade
0    3553
1    2213
2    1572
3    1548
Name: count, dtype: int64


In [30]:
df['bmi'] = df['weight_kg'] / ((df['height_cm'] / 100) ** 2)

df = df[(df['bmi'] >= 10) & (df['bmi'] <= 70)]
df = df[(df['height_cm'] >= 100) & (df['height_cm'] <= 220)]
df = df[(df['weight_kg'] >= 30) & (df['weight_kg'] <= 200)]

print(df['bmi'].describe())
print("Rows remaining:", len(df))

count    8777.000000
mean       26.080939
std         7.421038
min        12.869313
25%        20.865267
50%        24.447279
75%        29.615806
max        69.855358
Name: bmi, dtype: float64
Rows remaining: 8777


In [31]:
df['country_name'] = df['country'].map({241: 'Ghana', 155: 'South Africa'})
print(df['country_name'].value_counts())

country_name
Ghana           4928
South Africa    3849
Name: count, dtype: int64


In [32]:
# Age groups
def age_group(age):
    if age < 50:
        return '18-49'
    elif age < 60:
        return '50-59'
    elif age < 70:
        return '60-69'
    elif age < 80:
        return '70-79'
    else:
        return '80+'

df['age_group'] = df['age'].apply(age_group)
print(df['age_group'].value_counts())

age_group
50-59    3184
60-69    2284
70-79    1558
18-49    1119
80+       632
Name: count, dtype: int64


In [36]:
# Encode sex: Male = 1, Female = 0
df['sex_encoded'] = df['sex'].map({1: 1, 2: 0})
print(df['sex_encoded'].value_counts())

sex_encoded
0    4557
1    4220
Name: count, dtype: int64


In [37]:
# Age-sex interaction
df['age_sex_interaction'] = df['age'] * df['sex_encoded']
print(df['age_sex_interaction'].describe())

count    8777.000000
mean       28.635639
std        31.114450
min         0.000000
25%         0.000000
50%         0.000000
75%        58.000000
max       110.000000
Name: age_sex_interaction, dtype: float64


In [35]:
print(df['sex'].dtype)
print(df['sex'].unique())

int64
[2 1]


In [38]:
cols_to_drop = ['sbp1', 'sbp2', 'sbp3', 'dbp1', 'dbp2', 'dbp3', 
                'birth_month', 'birth_year']

df = df.drop(columns=cols_to_drop)
print(df.shape)
print(df.columns.tolist())

(8777, 31)
['id', 'country', 'sex', 'residence', 'age', 'height_cm', 'weight_kg', 'waist_cm', 'hip_cm', 'ever_tobacco', 'current_tobacco', 'ever_alcohol', 'recent_alcohol', 'avg_drinks_daily', 'vigorous_work', 'days_vigorous_work', 'moderate_work', 'walks_or_cycles', 'vigorous_leisure', 'moderate_leisure', 'fruit_servings', 'veg_servings', 'survey_weight', 'sbp_avg', 'dbp_avg', 'hypertension_grade', 'bmi', 'country_name', 'age_group', 'sex_encoded', 'age_sex_interaction']


In [39]:
df.to_csv('../docs/df_features.csv', index=False)
print("Saved successfully")

Saved successfully


### EDA

In [1]:
%pip install matplotlib seaborn

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Plot settings
plt.rcParams['figure.figsize'] = (10, 6)
sns.set_style('whitegrid')

df = pd.read_csv('../docs/df_features.csv')
print(df.shape)

ModuleNotFoundError: No module named 'matplotlib'